SI/Sem.07/Task05Pract7.cpp

```diff
@@ -0,0 +1,54 @@
+#include <iostream>
+const int MAX_LEN_OF_NUM = 10;
+int lenOfNum(int number){
+    int count = 0;
+    if(number == 0) return 1;
+    while(number != 0){
+        count++;
+        number /= 10;
+    }
+    return count;
+}
+void fillDigitsOfNumInArr(int * arr, int len, int number){
+    for(int i = len - 1; i >= 0; i--){
+        arr[i] = number % 10;
+        number /= 10;
+    }
+}
+int myPowF(int num, int power){
+    int newNum = 1;
+    for(int i = 0; i < power; i++){
+       newNum *= num;
+    }
+    return newNum;
+}
+int fromArrOfDigitsToInt(const int * arr, int len){
+    int resultNum = 0;
+    for(int i = 0; i < len; i++){
+       resultNum += arr[i] * myPowF(10, len-1-i);
+    }
+    return resultNum;
+}
+void swap(int& num1, int& num2){
+    int temp = num1;
+    num1 = num2;
+    num2 = temp;
+}
+void swapKDigit(int num1, int num2, int k){//task5-pract7
+    int lenOfNum1 = lenOfNum(num1);
+    int lenOfNum2 = lenOfNum(num2);
+    int digitsInNum1[MAX_LEN_OF_NUM];
+    int digitsInNum2[MAX_LEN_OF_NUM];
+    fillDigitsOfNumInArr(digitsInNum1, lenOfNum1, num1);
+    fillDigitsOfNumInArr(digitsInNum2, lenOfNum2, num2);
+    if(k <= lenOfNum1 && k <= lenOfNum2){
+        swap(digitsInNum1[k-1], digitsInNum2[k-1]);
```

Comment:
Write swap function yourself

Reply:
I have a swap function written myself.
32 to 36 line